code/modules/atmospherics/machinery/components/unary_devices/passive_vent.dm

```diff
@@ -20,36 +20,16 @@
 /obj/machinery/atmospherics/components/unary/passive_vent/process_atmos()
 	..()
 
-	var/active = FALSE
-
 	var/datum/gas_mixture/external = loc.return_air()
 	var/datum/gas_mixture/internal = airs[1]
-	var/external_pressure = external.return_pressure()
-	var/internal_pressure = internal.return_pressure()
-	var/pressure_delta = abs(external_pressure - internal_pressure)
-
-	if(pressure_delta > 0.5)
-		if(external_pressure < internal_pressure)
-			var/air_temperature = (external.temperature > 0) ? external.temperature : internal.temperature
-			var/transfer_moles = (pressure_delta * external.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			var/datum/gas_mixture/removed = internal.remove(transfer_moles)
-			external.merge(removed)
-		else
-			var/air_temperature = (internal.temperature > 0) ? internal.temperature : external.temperature
-			var/transfer_moles = (pressure_delta * internal.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			transfer_moles = min(transfer_moles, external.total_moles() * internal.volume / external.volume)
-			var/datum/gas_mixture/removed = external.remove(transfer_moles)
-			if(isnull(removed))
-				return
-			internal.merge(removed)
-
-		active = TRUE
-
-	active = internal.temperature_share(external, OPEN_HEAT_TRANSFER_COEFFICIENT) ? TRUE : active
-
-	if(active)
-		air_update_turf()
-		update_parents()
+
+	var/internal_volume_fraction = internal.volume/(internal.volume + external.volume)
+
+	external.merge(internal.remove_ratio(1))
+	internal.merge(external.remove_ratio(internal_volume_fraction))
```

Comment:
I feel we should test that this isn't a noop beforehand somehow

Reply:
You'd probably have to check temerature and each gas individually like it was doing before Dennok's comments, which might end up being just as or nearly as intensive.

Reply:
This is roughly what it would look like, don't know if it's better.  If so, I'd put this in a proc in gas_mixture as mix_volumes or something.

```
    var/active = FALSE
	active = active || abs(internal.return_pressure() - external.return_pressure) > 0.5
	active = active || abs(internal.return_temperature() - external.return_temperature()) > 5
	active = active || internal.gases.len != external.gases.len
	active = active || (internal.gases | external.gases).len == internal.gases.len
	for(var/gas_id in internal.gases)
		active = active || abs(internal.gases[gas_id][MOLES] - external.gases[gas_id][MOLES]) > 1
		if(active)
			break
	if(active)
		var/internal_volume_fraction = internal.volume/(internal.volume + external.volume)
		external.merge(internal.remove_ratio(1))
		internal.merge(external.remove_ratio(internal_volume_fraction))
				
		air_update_turf()
		update_parents()```

Reply:
for what `active = active || (internal.gases | external.gases).len == internal.gases.len`?

Reply:
Only purpose is to pick up if both mixes have the same number of gases but the gas types are different.  I can see now as well that the inside of the for loop is wrong.  Overall that code is just to show that you'd probably check all of:
1. Overall pressure difference
2. Overall temp difference
3. Individual gas differences

In that order to tell if it's a no-op.

Reply:
http://www.byond.com/docs/ref/index.html#/operator/|
As i understend, this check that we ALREADY has external.gases in internal.gases
we add external.gases in internal.gases and check that len of internal.gases dont changed

Reply:
The lines:
```	
        active = active || internal.gases.len != external.gases.len
	active = active || (internal.gases | external.gases).len == internal.gases.len```
were just to check that the two lists contain the same set of keys, with the first check being the most likely way they would be inequal, the second line catching an unlikely edge case.  Both don't care about ordering.

e.g. first line will catch (2,1,3) not being the same set as (1,2), but won't catch (1,2,3) and (4,2,1), but the second line will catch it.  I think why I did that way off the top of my head was because I don't know how byond.

That snippet was informal and just an example, I'm aware the details of it are wrong in some places.  I'll add a proc for mixing volumes in gas_mixture that actually works later today.

Reply:
if you want catch that list(1,2,3) and list(4,2,1) different in `(internal.gases | external.gases).len == internal.gases.len` you need use `!=`. Now it only check that external.gases alreadi in internal.gases. I say about this.

Reply:
oh ya my bad

Reply:
`active = active || (internal.gases ^ external.gases).len != 0` list XOR will give you any gases that aren't in both lists

Reply:
that being said make sure to profile whether it's worth it to actually do these checks